File: testdata/metrics/v2.ts
export function classify(value: number): number {
  if (value > 0) {
    for (let index = 0; index < value; index += 1) {
      if (index % 2 === 0) {
        value += 1;
      }
    }
  } else {
    value = value < -10 ? -10 : value;
  }

  switch (value) {
    case 1:
      return 10;
    case 2:
      return 20;
    default:
      return value;
  }
}

File: testdata/go/metrics/v2.go
package sample

func measure(values []int, in <-chan int, out chan<- int) int {
	total := 0

	for _, value := range values {
		if value > 0 {
			total += value
		} else if value < 0 {
			total -= value
		}
	}

	for index := 0; index < len(values); index++ {
		total += index
	}

	switch total {
	case 0:
		total = 1
	case 1:
		total = 2
	default:
		total += 1
	}

	select {
	case next := <-in:
		out <- next
	case out <- total:
	default:
	}

	return total
}
